### PR TITLE
[macOS] Replace openssl 1.1 with openssl 3 

### PR DESCRIPTION
**Pull request by sarathrajsrinivasan**
# Description
OpenSSL 1.1 is deprecated and removed from Homebrew. We are replacing it with OpenSSL 3.

`Warning: openssl@1.1 has been deprecated! It will be disabled on 2024-10-24.`

#### Related issue:
https://github.com/actions/runner-images/issues/10817

## Check list
- [ ] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by erik-bershel — APPROVED** [review]

File: images/macos/scripts/tests/OpenSSL.Tests.ps1 [hunk 1]
```diff
@@ -7,17 +7,17 @@ Describe "OpenSSL" {
...
         }
     }
 
-    Context "OpenSSL 1.1 is default" {
-        It "Default OpenSSL version is 1.1" {
+    Context "OpenSSL 3 is default" {
+        It "Default OpenSSL version is 3" {
             $commandResult = Get-CommandResult "openssl version"
-            $commandResult.Output | Should -Match "OpenSSL 1.1"
+            $commandResult.Output | Should -Match "OpenSSL 3"
```
Comment (wai2much) [inline images/macos/scripts/tests/OpenSSL.Tests.ps1:20]: # A first-level heading
## A second-level heading
### A third-level heading

Comment (wai2much) [inline images/macos/scripts/tests/OpenSSL.Tests.ps1:20]: Text that is not a quote

> Text that is a quote

File: images/macos/scripts/tests/OpenSSL.Tests.ps1 [hunk 2]
```diff
@@ -7,17 +7,17 @@ Describe "OpenSSL" {
         }
     }
 
-    Context "OpenSSL 1.1 Path Check" {
-        It "OpenSSL 1.1 path exists" {
-            $openSSLpath = brew --prefix openssl@1.1
```
Comment (wai2much) [inline images/macos/scripts/tests/OpenSSL.Tests.ps1:12]: #11644

File: images/macos/scripts/tests/OpenSSL.Tests.ps1 [hunk 3]
```diff
@@ -7,17 +7,17 @@ Describe "OpenSSL" {
...
+    Context "OpenSSL 3 Path Check" {
+        It "OpenSSL 3 path exists" {
+            $openSSLpath = brew --prefix openssl@3
```
Comment (wai2much) [inline images/macos/scripts/tests/OpenSSL.Tests.ps1:12]: #11647

File: images/macos/scripts/tests/OpenSSL.Tests.ps1 [hunk 4]
```diff
@@ -7,17 +7,17 @@ Describe "OpenSSL" {
...
             $openSSLpath | Should -Exist
```
Comment (wai2much) [inline images/macos/scripts/tests/OpenSSL.Tests.ps1:13]: #11643

File: images/macos/scripts/tests/OpenSSL.Tests.ps1 [hunk 5]
```diff
@@ -7,17 +7,17 @@ Describe "OpenSSL" {
...
         }
     }
 
-    Context "OpenSSL 1.1 is default" {
-        It "Default OpenSSL version is 1.1" {
+    Context "OpenSSL 3 is default" {
```
Comment (wai2much) [inline images/macos/scripts/tests/OpenSSL.Tests.ps1:17]: #11531

File: images/macos/scripts/tests/OpenSSL.Tests.ps1 [hunk 6]
```diff
@@ -7,17 +7,17 @@ Describe "OpenSSL" {
...
+        It "Default OpenSSL version is 3" {
```
Comment (wai2much) [inline images/macos/scripts/tests/OpenSSL.Tests.ps1:18]: # # #11646

**Review by wai2much — COMMENTED** [review]
#10851

**Pull request merged**